src/server/middleware/pwa/validation.js

```diff
@@ -14,24 +14,149 @@
  * permissions and limitations under the License.
  */
```

Comment:
This might be the start of a very lightweight validation library as alternative to other libraries Ive seen already implementing this similar logic like https://github.com/25th-floor/spected#advanced-example. This doesnt seem especially PWA specific this logic and could be used elsewhere as in for other appConfig validations going forward? How about moving this to something like `src/server/validate-shape/validations.js` perhaps?

Reply:
@mtomcal I can def see that this can be used by other parts of the app when it comes to checking/validation - could we give more thought to the idea first before moving it? Maybe only move over the helper functions with some preset shapes used - `const { validateWebManifest, isString, createIsArrayOf, createValidator } = require('../')`?

@JAdshead A `WebManifest` TS type would certainly make that runtime validation obsolete, I think the main perk of the config is it's meant to log what's expected from the manifest object shape when devving so the user can fix it on the go.

Reply:
@JAdshead I think runtime validation is essential however, the generic nature of these seems like they need to be organized differently. We will come accross more situations where we want to provide detailed validations for any new App Configuration shapes and these will come in handy unless we remove these in favor of library that specializes in this kind of validation.

@Francois-Esquire Would you investigate into lightweight validation library alternatives we have used previously (like ajv) or some other library that would save us from maintaining our own validation tooling?

Reply:
We decided as a team to punt the validation library conversation until later. Resolving this convo.